### PR TITLE
kirkwood/mvebu: refresh kernel config

### PR DESCRIPTION
**Pull request by mans0n**
kirkwood: refresh kernel config
mvebu: refresh kernel config
mvebu: move boolean symbols from modules.mk to config

### CODE REVIEW

File: target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c [hunk 1]
```diff
@@ -458,6 +458,49 @@ static struct mtd_part_parser uimage_fonfxc_parser = {
 	.parse_fn = mtdsplit_uimage_parse_fonfxc,
 };
 
+
+/**************************************************
+ * ipTIME (Kirkwood devices)
```
Comment (mans0n) [inline target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c:463]: I'll happily adopt `openwrt,uimage` if the [mtdsplit_uimage PR](https://patchwork.ozlabs.org/project/openwrt/cover/20201130221935.11127-1-bjorn@mork.no/) get merged.

File: target/linux/kirkwood/patches-5.4/997-device_tree_cmdline.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
+--- a/drivers/of/fdt.c
++++ b/drivers/of/fdt.c
```
Comment (adschm) [inline target/linux/kirkwood/patches-5.4/997-device_tree_cmdline.patch:2]: A (short) comment or header on that patch could be helpful.

File: target/linux/mvebu/modules.mk [hunk 3]
```diff
@@ -2,11 +2,7 @@ define KernelPackage/linkstation-poweroff
   SUBMENU:=$(OTHER_MENU)
   DEPENDS:=@TARGET_mvebu
   TITLE:=Buffalo LinkStation power off driver
-  KCONFIG:= \
-	CONFIG_POWER_RESET=y \
-	CONFIG_POWER_RESET_LINKSTATION \
-	CONFIG_POWER_RESET_QNAP=n
-
+  KCONFIG:=CONFIG_POWER_RESET_LINKSTATION
```
Comment (adschm) [inline target/linux/mvebu/modules.mk:5]: On a quick look, I wonder whether this and the kernel config changes should be in a separate commit ...

File: target/linux/kirkwood/patches-5.4/997-device_tree_cmdline.patch (hunk 2 above)
Comment (mans0n) [reply target/linux/kirkwood/patches-5.4/997-device_tree_cmdline.patch:2]: You're right, I'll add it.

File: target/linux/mvebu/modules.mk (hunk 3 above)
Comment (mans0n) [reply target/linux/mvebu/modules.mk:5]: Yeah, that would be better.

**Comment by mans0n** [comment]
Made separate patches for `997-device_tree_cmdline.patch` and `CONFIG_POWER_RESET`.

File: target/linux/kirkwood/patches-5.4/997-device_tree_cmdline.patch (hunk 2 above)
Comment (mans0n) [reply target/linux/kirkwood/patches-5.4/997-device_tree_cmdline.patch:2]: Maybe we'd better move the patch to generic (as now three targets use it), but I'm not sure if it's safe...

**Comment by mans0n** [comment]
Added small change to fix warning messages.

File: target/linux/kirkwood/config-5.4 [hunk 4]
```diff
@@ -1,22 +1,6 @@
 CONFIG_ALIGNMENT_TRAP=y
 CONFIG_ARCH_32BIT_OFF_T=y
 CONFIG_ARCH_CLOCKSOURCE_DATA=y
-CONFIG_ARCH_HAS_BINFMT_FLAT=y
```
Comment (CHKDSK88) [inline target/linux/kirkwood/config-5.4:4]: Should be kernel refresh different patch?

Comment (mans0n) [reply target/linux/kirkwood/config-5.4:4]: This seams to be the result of a8fb12a7d62c and I feel it deserves a separate patch.
Or do you mean consolidating two refresh commits?

Comment (CHKDSK88) [reply target/linux/kirkwood/config-5.4:4]: So, it's treewide. Maybe it should be done for all targets in one commit...

@adschm 
What is Your opinion about it?

Comment (mans0n) [reply target/linux/kirkwood/config-5.4:4]: Hmm... It seems I need to build a whole toolchain to run `make kernel_oldconfig` and it's out of my scope to do that for every target...

Comment (adschm) [reply target/linux/kirkwood/config-5.4:4]: I'm not sure what's exactly the question, as the general refresh removing the ARCH symbols is already in separate commits.

But I think it's good to have them separate indeed.

**Comment by CHKDSK88** [comment]
I think about one treewide commit for all targets. But if now is ok, please forgot about it.

File: target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c (hunk 1 above)
Comment (mans0n) [reply target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c]: Finally merged! 3f36dffbdee0

**Comment by mans0n** [comment]
Rebased.
- replaced extra mtdsplit code to `openwrt,uimage` in DTS
- renamed firmware-utils tool
- fixed incorrect `IMAGE_SIZE`

**Comment by mans0n** [comment]
Your reviews were really helpful, but I've decided to abandon this, sorry for it.
@adschm Would you still like to take the kernel config patches?

**Comment by adschm** [comment]
I will take the refresh patches, and would not have liked the other change very much anyway (since it might not be relevant for buildbots, but it is for local builds).
Thanks.

**Comment by adschm** [comment]
I get a lot of prompts when running kernel_oldconfig for cortexa53. Since I'm not really keen on analysing this right now, I will only take the kirkwood refresh for now. I'm not reopening since I'm not sure whether you are interested in continuing this or will leave it as well.

**Comment by mans0n** [comment]
Fine with me. I don't know much about mvebu 64-bit, so (sadly) that would be out of my range. I guess kernel 5.10 bump will come soon, config refresh will be handled then.
